mlflow/gateway/client.py

```diff
@@ -286,4 +294,18 @@ def query(self, route: str, data: Dict[str, Any]):
 
         query_route = assemble_uri_path([MLFLOW_GATEWAY_ROUTE_BASE, route, MLFLOW_QUERY_SUFFIX])
 
-        return self._call_endpoint("POST", query_route, data).json()
+        try:
+            return self._call_endpoint("POST", query_route, data).json()
+        except MlflowException as e:
+            if re.search(r"timeout", str(e), re.IGNORECASE):
+                timeout_message = (
```

Comment:
Does requests have an error like `TimeoutError`?

Reply:
+1 would be better if we can catch a specific exception

Reply:
`requests.exceptions.Timeout` :) making the change!